mifosng-android/src/main/java/com/mifos/mifosxdroid/passcode/PassCodeActivity.java

```diff
@@ -84,15 +83,17 @@ public void savePassCode() {
         }
     }
 
-    @OnClick(R.id.btn_login)
-    public void loginUsingPassCode() {
+    @Override
+    public void passCodeEntered(String passcode) {
 
         if (!isInternetAvailable()) {
+            passCodeView.clearPasscodeField();
             return;
         }
 
         if (counter == 3) {
-            Toaster.show(clRootview, R.string.incorrect_passcode_more_than_three);
+            Toast.makeText(getApplicationContext(), R.string.incorrect_passcode_more_than_three,
+                    Toast.LENGTH_SHORT).show();
             PrefManager.clearPrefs();
```

Comment:
don't clear the Preference it's causing Runtime crash because Dagger is initializing the Retrofit and retrofit need base URL that you are clearing that doesn't make any sense.

Instead of clear the passcode and tell the user, log in again.